src/editor/ExpirationPanel.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { ToggleControl, DatePicker } from '@wordpress/components';
+import { isInTheFuture } from '@wordpress/date';
+import { useEffect, useState } from '@wordpress/element';
+
+const ExpirationPanel = ( {
+	expiration_date = null,
+	postStatus,
+	onMetaFieldChange,
+	createNotice,
+	removeNotice,
+} ) => {
+	const [ noticeId, setNoticeId ] = useState( false );
+	useEffect( () => {
+		if ( expiration_date && ! isInTheFuture( expiration_date ) ) {
+			if ( noticeId ) {
+				return;
+			}
+			createNotice(
+				'warning',
+				sprintf(
+					/* translators: %s is the expiration date */
+					__(
+						'This prompt has expired on %s and has been reverted to draft. Publishing it will reset the expiration date.',
+						'newspack-plugin'
+					),
+					new Date( expiration_date ).toLocaleDateString()
+				),
+				{
+					isDismissible: false,
+				}
+			).then( ( { notice } ) => {
+				setNoticeId( notice.id );
+			} );
+		}
+	}, [ expiration_date ] );
+
+	useEffect( () => {
+		if ( postStatus === 'publish' && noticeId ) {
+			removeNotice( noticeId );
+			createNotice(
+				'info',
+				__(
+					'This prompt has been published. The expiration date has been removed.',
+					'newspack-plugin'
+				)
+			);
+			// This is just for quicked feedback, the actual meta field deletion will
+			// happen on the backend.
+			onMetaFieldChange( { expiration_date: null } );
+		}
+	}, [ postStatus ] );
+
+	return (
+		<>
+			<ToggleControl
```

Comment:
Nit: The setting seems a bit bare. It might be a good idea to add some helper text here to describe what the setting will do.

![Screenshot 2024-05-22 at 13 34 00](https://github.com/Automattic/newspack-popups/assets/17905991/e635a880-f9bd-49af-9876-d952fcb60ccb)

Feel free to ignore this one though as I think it might be obvious enough without the helper text.

Reply:
Good idea! Added in 5cf76483d8991112d343b00cb6af66c9913950ec